src/Microsoft.NetCore.Analyzers/Core/Runtime/UseToLowerInvariantOrToUpperInvariant.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using Analyzer.Utilities;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Runtime
+{
+    public abstract class UseToLowerInvariantOrToUpperInvariantAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string RuleId = "CA_IHaveNoIdeaHowTheseAreNumbered";
+
```

Comment:
@mavasani how do I deal with rule id? I don't know how you guys number these things.
The diagnostics are complaining about it and it's causing the builds to fail.

Reply:
We need to decide if this gets a new rule ID or can be combined with any existing rule ID. If it is a new rule ID, then you identify the category of the rule and pick the next available rule ID from the rule range by category: https://github.com/dotnet/roslyn-analyzers/blob/master/src/Utilities/Compiler/DiagnosticCategoryAndIdRanges.txt

As a followup when the PR gets merged, you should also file a doc bug to add documentation for it. For example, for globalization rule, you will file a content bug [here](https://docs.microsoft.com/en-us/visualstudio/code-quality/globalization-warnings?view=vs-2017) and also a follow-up bug in the repo to update the analyzer hyperlink to the newly created doc page.

Reply:
@mavasani Shall it be a new rule ID?

Reply:
@mavasani I don't think we need to get this analyzer merged with an existing rule. Let's use a brand new rule ID.

@TKharaishvili you can look at other merged PRs for guidance. For example, [here is a PR I recently merged](https://github.com/dotnet/roslyn-analyzers/pull/3641/files). Search for any instances of `CA2016` so you know which files you need to modify and you can replicate the same here.